### PR TITLE
Allow the ability to connect to Postgres via an SSH tunnel

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

File: flow/connectors/postgres/ssh_wrapped_pool.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	var err error
+
+	// Establish an SSH connection
+	swpp.sshClient, err = ssh.Dial("tcp", swpp.sshServer, swpp.sshConfig)
+	if err != nil {
+		return err
+	}
+	logrus.Info("SSH connection established")
+	// Automatically pick an available local port
+	localListener, err := net.Listen("tcp", "localhost:0")
```
Comment (serprex) [inline flow/connectors/postgres/ssh_wrapped_pool.go:98]:
```suggestion
	localListener, err := net.Listen("unix", tmpFile)
```
Did you evaluate unix domain sockets? They have better perf: https://redis.io/docs/management/optimization/benchmarks

> When the server and client benchmark programs run on the same box, both the TCP/IP loopback and unix domain sockets can be used. Depending on the platform, unix domain sockets can achieve around 50% more throughput than the TCP/IP loopback (on Linux for instance). The default behavior of redis-benchmark is to use the TCP/IP loopback.

Comment (iskakaushik) [reply flow/connectors/postgres/ssh_wrapped_pool.go]: @serprex I reworked this to just modify the `Dial` func, this seems to be what the author recommends as well: https://github.com/jackc/tern/blob/4ca45ec386dd3b7c2b174a16d9a9522bac392b26/ssh_tunnel.go#L21

File: flow/connectors/utils/ssh.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	if len(authMethods) == 0 {
+		return nil, fmt.Errorf("no authentication methods provided")
+	}
+
+	return &ssh.ClientConfig{
+		User:            user,
+		Auth:            authMethods,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
```
[the remark this answers is not in the document]
Comment (serprex) [reply flow/connectors/utils/ssh.go:46]: This should be configured. We can get host key from our existing ssh connection with them. Callback only needs to check host key, not ip

File: flow/connectors/postgres/ssh_wrapped_pool.go (hunk 1 above)
Comment (serprex) [reply flow/connectors/postgres/ssh_wrapped_pool.go]: Dial's first parameter can be `"unix"`

https://github.com/jackc/tern/blob/4ca45ec386dd3b7c2b174a16d9a9522bac392b26/ssh_tunnel.go#L47

Comment (iskakaushik) [reply flow/connectors/postgres/ssh_wrapped_pool.go]: Got it!

**Pull request merged**